src/charm.py

```diff
@@ -169,6 +173,11 @@ def _restart_flask(self) -> None:
         except CharmConfigInvalidError as exc:
             self._update_app_and_unit_status(ops.BlockedStatus(exc.msg))
 
+    def _on_update_status(self, _: ops.HookEvent) -> None:
+        """Handle the update-status event."""
+        if self._database_migration.get_status() == "FAILED":
+            self._restart_flask()
```

Comment:
Will `_restart_flask` remove the status file?

Reply:
If the database migration run succeeded the status file will be set to "COMPLETED".

Reply:
I meant if there's a `FAILED` status stored. Note that `DatabaseMigration.run` will only execute the migration if the status is `PENDING`

Reply:
You're absolutely correct; thank you for pointing that out. I've addressed the issue and implemented additional tests to prevent similar problems.